### ISSUE TITLE
0.5.3 to current master: Still unable to continue training after original iteration reached. And checkpoint export feature now missing

### ISSUE THREAD

**Issue by momentmal** [MrNeRF/LichtFeld-Studio#1909]
Since I observed a similar issue as in #1250,  #1648  and #1702 i switched from my 0.5.3 build to current master (8e1a6ac2) and rebuilt for Windows.

Seems like in the hopes of having the ability to now resume from a checkpoint to go beyond the trained 30.000 default iterations, it seems i have no possibility to save one anymore.
I did a test training up to 5000 iterations, then training stops and I see no feature anymore that allows me to export a checkpoint .resume file.

I guess It either moved somewhere I am overlooking or it is a bug?

The usecase is pretty simple, I want to be able to increase iterations after i reached the point of iterations i have set for first training run, e.g. the default 30k when I realize it would benefit from more iterations.

This neither works for me in any way with the GUI on 0.5.3 nor on current master.
The only way I could find was running LFS in headless mode which does not allow me to see visual progress while continuing training.
Also it requires me to have a checkpoint, which i apparently I am not able to create myself manually anymore

`LichtFeld-Studio.exe --headless -d "\frames-sharpest-350\export-1" -o "\frames-sharpest-350\export-1\output" --resume "\frames-sharpest-350\export-1\output\checkpoints\checkpoint_30000.resume" -i 120000`

**Comment by MrNeRF** [comment]
There is the new project format .licht which stores checkpoints and can resume them.  The old checkpoints are not created anymore they can be still imported.

Training extension will be added  later though I doubt a large positive effect as the main parameters have converged towards almost off per scheduler. 

I don't recommend setting -i manually unless you really know what you are doing.

**Comment by momentmal** [comment]
Thanks for the quick feedback! I will try out those new .licht files for my usecase. 👌

> unless you really know what you are doing.

I'm rather new in the splats field (yet familiar with classic photogrammetry workflows), I felt like having a flexible iteration goal is one of the main influences on the final quality, but you probably have more experience in this 😎

I remember Jörg Kahlhöfer [training up to 162k](https://youtu.be/OzUxL_UDMTk?si=5QSi_gkrCYYctd4h&t=1508) iterations (in a different software) while observing if quality improves, I think his results speak for themselves, but maybe workflows differ vastly between softwares (and people) and depending on the scene captured.

> the new format .licht which stores checkpoints and can resume them. 
> Training extension will be added later 

So resuming is currently only possible if the original iteration goal set for the training is not yet reached?
That would mean iteration goal is "set in stone" once and for all for an individual training process and cannot be increased at the moment?
It sounds like a useful feature if training extension would be added 👌

The aequivalent approach to Jörgs workflow in LFS  for now then probably would be to set a rather high iteration goal and stop training once the result looks satisfying and then save the scenefile to leave the option open to continue training further at a later point if necessary. Of course photo input quality and the other training settings also have great impact and need to be optimized accordingly.

**Comment by momentmal** [comment]
Some more testing confirms my assumptions - i can continue training only when loading a project with unfinished training.

There currently appears to be no way to modifiy training iteration target once training has started.

I will look forward to:
> Training extension will be added later 

🙃 - I assume you don't need a seperate issue / feature request created for it as you have it already in future planning somewhere?

As you made clear that current behavior is known and intentional, do you want me to close the issue for now?

On a slight tangent: I feel like putting checkpoints all into the .licht project file is making checkpoint storage a bit less transparent (but gives us a cleaner file structure). I added "recently modified" dates to my file view to be able to see when the last checkpoint was packed into the project file. A named .ply / .resume file was more telling, as it included the number of training steps already saved. It still might be worth the tradeoff, as it definitely keeps file structure cleaner.

I think having ability to modify / extend training target after the fact will be beneficial, until then I will keep experimenting and enjoying your nice piece of sofware 😎

Created another set of imagery today, excited to test it. 🤩

**Comment by MrNeRF** [comment]
You can leave it open - It will arrive some time but I think you overestimate the effect. There are other ways to get "there".
The learning rates are so small towards the end that almost nothing will happen anymore just mainly overfitting.

Stay tuned. There will be some ui around the .licht format. Things aren't happening over night. A project format is a extremely huge and impactful project.

**Comment by momentmal** [comment]
> You can leave it open - It will arrive some time but I think you overestimate the effect. There are other ways to get "there". The learning rates are so small towards the end that almost nothing will happen anymore just mainly overfitting.
> 
> Stay tuned. There will be some ui around the .licht format. Things aren't happening over night. A project format is a extremely huge and impactful project.

Thanks! I will keep testing & learning on how to affect results 👌 Currently having some trouble building current master on Windows.
Not sure if it is helpful to create an issue for it or it rather puts overhead on you?

No worries about timeframes, I am patient (only getting warmed up anyways) and I think the project format is a good idea (sounds like a huge project indeed!) - so don't take any of my comments as "pressuring requests", just remarks with intention to improve the software ^^
